pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -341,7 +345,8 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
 
         if not weights_only:
             # dump callbacks
-            checkpoint["callbacks"] = self.trainer._call_callbacks_on_save_checkpoint(checkpoint)
+            checkpoint["callbacks_state_dict"] = self.trainer._call_callbacks_state_dict()
+            checkpoint["callbacks_deprecated_hook_states"] = self.trainer._call_callbacks_on_save_checkpoint(checkpoint)
```

Comment:
this needs to be at the end right? otherwise this is not operating on the full checkpoint dict

Reply:
yes, I hesitated a bit to move the order around but it shouldn't matter if `checkpoint` gets additional entries (which it should be getting) before it gets passed to `Callbacks.on_save_checkpoint()` - will move this call down